docs/illixr_plugins.md

```diff
@@ -1,85 +1,193 @@
-        [_fast pose_][37] every time a new IMU sample arrives using
-        the GTSAM library ([upstream][11]).
+    Integrates over all [_IMU_][36] samples since the last published [_SLAM_][39] pose to provide a
+        [_fast pose_][37] every time a new IMU sample arrives using the GTSAM library ([upstream][11]).
+
+    Topic details:
+
+    -   *Publishes* `imu_raw_type` on `imu_raw` topic.
+    -   Synchronously *reads/subscribes* to `imu_cam_type` on `imu_cam` topic.
+    -   Asynchronously *reads* `imu_integrator_input` on `imu_integrator_input` topic.
 
 -   [`pose_prediction`][17]:
-    Uses the latest IMU value to predict a pose for a future point in time.
+    Uses the latest [_IMU_][36] value to predict a [_pose_][37] for a future point in time.
+    Implements the `pose_prediction` service (defined in `common`),
+        so poses can be served directly to other plugins.
+
+    Topic details:
+
+    -   Asynchronously *reads* `pose_type` on `slow_pose` topic,
+            but it is only used as a fallback.
+    -   Asynchronously *reads* `imu_raw` on `imu_raw` topic.
+    -   Asynchronously *reads* `pose_type` on `true_pose` topic,
+            but it is only used if the client asks for the true pose.
+    -   Asynchronously *reads* `time_type` on `vsync_estimate` topic.
+        This tells `pose_predict` what time to estimate for.
 
 -   [`gldemo`][5]:
-    Renders a static scene (into left and right [_eye buffers_][34]) given the pose
+    Renders a static scene (into left and right [_eye buffers_][34]) given the [_pose_][37]
         from `pose_prediction`.
 
+    Topic details:
+
+    -   *Publishes* `rendered_frame` on `eyebuffer` topic.
+    -   Asynchronously *reads* `time_type` on `vsync_estimate` topic.
+
 -   [`timewarp_gl`][6]:
-    [Asynchronous reprojection][34] of the [_eye buffers_][34].
+    [Asynchronous reprojection][35] of the [_eye buffers_][34].
+    The timewarp ends just after [_vsync_][34], so it can deduce when the next vsync will be.
 
--   [`debugview`][7]: Renders a frame for debug information.
+    Topic details:
 
--   [`audio`][8]:
-    ***TODO.***
-    By default this is disabled, since it requires an NVIDIA GPU.
+    -   *Calls* `pose_prediction`.
+    -   Asynchronously *reads* `rendered_frame` on `eyebuffer` topic.
+    -   *Publishes* `time_type` on `mtp` topic, which is an estimate of the _nominal_ motion-to-photon latency.
+    -   *Publishes* `time_type` on `warp_frame_age` topic.
+    -   *Publishes* `type_type` on `vsync_estimate`.
+    -   *Publishes* `hologram_input` on `hologram_in` topic.
+    -   *Publishes* `texture_pose` on `texture_pose` topic if `ILLIXR_OFFLOAD_ENABLE` is set in the env.
 
--   [`hologram`][9]:
-    ***TODO.***
-    By default this is disabled, since it requires an NVIDIA GPU.
+-   [`debugview`][7]:
+    Renders incoming [_frames_][34] from the graphics pipeline for debugging live executions of the application.
+
+    Topic details:
```

Comment:
Should `imu_raw` be called `fast_pose` here?
```cpp
   -   Asynchronously *reads* `imu_raw` on `imu_raw` topic.
```
```cpp
    debugview(std::string name_, phonebook *pb_)
        : threadloop{name_, pb_}
        , sb{pb->lookup_impl<switchboard>()}
        , pp{pb->lookup_impl<pose_prediction>()}
        , _m_slow_pose{sb->subscribe_latest<pose_type>("slow_pose")}
        , _m_fast_pose{sb->subscribe_latest<imu_raw_type>("imu_raw")}
        //, glfw_context{pb->lookup_impl<global_config>()->glfw_context}
    {}
```

Reply:
`fast_pose` could work, but the problem is that `imu_raw` also includes the last two (biased) IMU readings, so it's not just the pose. Open to naming suggestions.

Reply:
Right. Pose prediction uses all of `imu_raw` but debugview only uses the pose part of it, and we don't anticipate using the biases in debugview. So we can point that out. But I'm still not sure whether we should call the pose part `fast_pose` because it is not exactly a pose type, but rather has the position and quaternion stored separately.